src/llm_vm/completion/data_synthesis.py

```diff
@@ -40,7 +40,7 @@ def data_synthesis(self, optimizer, prompt, response, openai_key=None, completio
         final_prompt = "Generate 1 json similar to the one below. \n" + final_prompt
 
         while len(datapoints) < self.examples_to_generate:
-            datapoint = self.generate_example(final_prompt, openai_key, completion=completion)
+            datapoint = self.generate_example(optimizer, final_prompt, openai_key, completion=completion, **kwargs)
```

Comment:
this is the main fix right?

Reply:
passing `optimizer` here to access `optimizer.big_model` below